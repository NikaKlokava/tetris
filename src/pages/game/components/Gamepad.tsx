import { memo } from "react";
import classnames from "classnames";
import cl from "./styles.module.css";
import Emitter from "../../../shared/emitter/EventEmitter";

export const Gamepad = memo(() => {
  const handleBtnClick = (type: ButtonTypes) => {
    Emitter.emit(type);
  };
  return (
    <div className={cl.gamepad_wrapper}>
      <div className={cl.move_buttons}>
        <button
          className={classnames(cl.gamepad_btn, cl.btn_left)}
          onClick={() => handleBtnClick("left")}
        ></button>
        <button
          className={classnames(cl.gamepad_btn, cl.btn_right)}
          onClick={() => handleBtnClick("right")}
        ></button>
        <button
          className={classnames(cl.gamepad_btn, cl.btn_down)}
          onClick={() => handleBtnClick("down")}
        ></button>
      </div>
      <div className={cl.rotate_button}>
        <button
          className={classnames(cl.gamepad_btn, cl.btn_rotate)}
          onClick={() => handleBtnClick("rotate")}
        ></button>
      </div>
    </div>
  );
});
