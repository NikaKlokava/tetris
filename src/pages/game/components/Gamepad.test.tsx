import renderer from "react-test-renderer";
import { Gamepad } from "./Gamepad";

describe("Test the Gamepad component", () => {
  test("The Gamepad renders correctly", () => {
    const gamepadSnap = renderer.create(<Gamepad />).toJSON();
    expect(gamepadSnap).toMatchSnapshot();
  });
});
