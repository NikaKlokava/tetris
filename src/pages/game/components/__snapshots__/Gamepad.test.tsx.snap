// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Gamepad component The Gamepad renders correctly 1`] = `
<div
  className="gamepad_wrapper"
>
  <div
    className="move_buttons"
  >
    <button
      className="gamepad_btn btn_left"
      data-testid="gamepad-btn"
      onClick={[Function]}
    />
    <button
      className="gamepad_btn btn_right"
      data-testid="gamepad-btn"
      onClick={[Function]}
    />
    <button
      className="gamepad_btn btn_down"
      data-testid="gamepad-btn"
      onClick={[Function]}
    />
  </div>
  <div
    className="rotate_button"
  >
    <button
      className="gamepad_btn btn_rotate"
      data-testid="gamepad-btn"
      onClick={[Function]}
    />
  </div>
</div>
`;
